remind.py

```diff
@@ -0,0 +1,118 @@
+import jinja2
+from smtplib import SMTP
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+from functions import generate_html_file, get_osp_version, has_blockers
+
+
+def run_remind(config, blockers, server, header):
+
+	# get set of all owners in blocker file
+	owner_list = []
+	for job in blockers:
+		owners = blockers[job].get('owners', False)
+		if not owners:
+			continue
+		owner_list.extend(owners)
+	owner_set = set(owner_list)
+
+	# find each job with no blockers including the owner and send email with agg'd list
+	for owner in owner_set:
+		rows = []
+		for job_name in blockers:
+			owners = blockers[job_name].get('owners', [])
+			if owner not in owners:
+				continue
+			osp_version = get_osp_version(job_name)
+
+			# get all relevant info from jenkins
+			try:
+				job_info = server.get_job_info(job_name)
+				job_url = job_info['url']
+				lcb_num = job_info['lastCompletedBuild']['number']
+				build_info = server.get_build_info(job_name, lcb_num)
+				build_actions = build_info['actions']
+				component = [action['text'] for action in build_actions if 'COMPONENT' in action.get('text', '')]
+
+				# if build was testing specific component, find next most recent build not testing a component
+				while component != []:
+					lcb_num = lcb_num - 1
+					build_info = server.get_build_info(job_name, lcb_num)
+					build_actions = build_info['actions']
+					component = [action['text'] for action in build_actions if 'COMPONENT' in action.get('text', '')]
+
+				lcb_url = build_info['url']
+				lcb_result = build_info['result']
+				compose = [action['text'][13:-4] for action in build_actions if 'core_puddle' in action.get('text', '')]
+
+				# No compose could be found; likely a failed job where the 'core_puddle' var was never calculated
+				if compose == []:
+					compose = "Could not find compose"
+				else:
+					compose = compose[0]
+
+			except Exception as e:
+
+				# No "Last Completed Build" found
+				if job_info['builds'] == []:
+					lcb_num = None
+					compose = "N/A"
+					lcb_url = None
+					lcb_result = "NO_KNOWN_BUILDS"
+
+				# Unknown error, skip job
+				else:
+					print("Jenkins API call error on job {}: {}".format(job_name, e))
+					continue
+
+			# only care about jobs with UNSTABLE or FAILURE status that have no blockers
+			if lcb_result == "UNSTABLE" or lcb_result == "FAILURE":
+				if not has_blockers(blockers, job_name):
+					row = {
+						'osp_version': osp_version,
+						'job_name': job_name,
+						'job_url': job_url,
+						'lcb_num': lcb_num,
+						'lcb_url': lcb_url,
+						'compose': compose,
+						'lcb_result': lcb_result,
+					}
+					rows.append(row)
+
+		# if no rows were generated, all jobs belonging to owner are already triaged
+		if rows != []:
```

Comment:
Hmm, an else here would make things better consumable. Currently you don't know if things worked if you only had owners against jobs that passed..

Reply:
Good idea, what did you have in mind for `else` behavior? Maybe just a print to stdout?

Reply:
Yep, just stdout, perhaps with a friendly message that there's nothing to remind you about, but, you could still generate a report with the same params if you wish..